internal/model/db_mysql.go

```diff
@@ -0,0 +1,196 @@
+// Copyright 2021 The golang.design Initiative Authors.
+// All rights reserved. Use of this source code is governed
+// by a MIT license that can be found in the LICENSE file.
+//
+// Originally written by Mai Yang <maiyang.me>.
+
+package model
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"time"
+
+	_ "github.com/go-sql-driver/mysql"
+	"github.com/jmoiron/sqlx"
+)
+
+// Store is persistent storage that provides a group of operations
+// to interact with the underlying database.
+type Store struct {
+	sqlxDB *sqlx.DB
+}
+
+// NewDB parses a given DSN and returns a DB instance for
+// further operations. It returns an error if the database
+// instance is not able to connect.
+func NewDB(dsn string) (*Store, error) {
+	db, err := sqlx.Open("mysql", dsn)
+	if err != nil {
+		fmt.Printf("connect server failed, err:%v\n", err)
+		return nil, err
+	}
+	db.SetMaxOpenConns(120)
+	db.SetMaxIdleConns(50)
+	return &Store{sqlxDB: db}, nil
+}
+
+func (db Store) Close() (err error) {
+	err = db.sqlxDB.Close()
+	if err != nil {
+		err = fmt.Errorf("failed to close database: %w", err)
+	}
+	return
+}
+
+// StoreAlias stores a given short alias with the given link if not exists
+func (db Store) StoreAlias(ctx context.Context, r *Redirect) error {
+	now := time.Now().UTC()
+	query, args, err := sqlx.In(`
+INSERT INTO collink (alias, kind, url, private, created_at, updated_at)
+VALUES(?, ?, ?, ?, ?, ?)
+`, r.Alias, r.Kind, r.URL, r.Private, now, now)
+	_, err = db.sqlxDB.ExecContext(ctx, query, args...)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// UpdateAlias updates the link of a given alias
+func (db Store) UpdateAlias(ctx context.Context, red *Redirect) error {
+	query, args, err := sqlx.In(`
+UPDATE collink
+SET url=?
+WHERE alias=?
+`, red.URL, red.Alias)
+	_, err = db.sqlxDB.ExecContext(ctx, query, args...)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// DeleteAlias deletes a given short alias if exists
+func (db Store) DeleteAlias(ctx context.Context, a string) error {
+	query, args, err := sqlx.In(`
+DELETE FROM collink WHERE alias=?
+`, a)
+	_, err = db.sqlxDB.ExecContext(ctx, query, args...)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// FetchAlias reads a given alias and returns the associated link
+func (db Store) FetchAlias(ctx context.Context, a string) (*Redirect, error) {
+	query, args, err := sqlx.In(`
+SELECT alias,
+       kind,
+       url,
+       private
+FROM collink
+WHERE alias=?
+`, a)
+	red := []*Redirect{}
+	err = db.sqlxDB.SelectContext(ctx, &red, query, args...)
+	if err != nil {
+		return nil, err
+	}
+	if len(red) > 0 {
+		return red[0], nil
+	}
+	return nil, sql.ErrNoRows
```

Comment:
```suggestion
	if len(red) == 0 {
		return nil, sql.ErrNoRows
	}
	return red[0], nil
```

错误先返回，使代码前后一致（if err != nil）